guiEditor/src/guiEditor.ts

```diff
@@ -0,0 +1,83 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { GlobalState } from './globalState';
+import { WorkbenchEditor } from './workbenchEditor';
+import { Popup } from "./sharedComponents/popup";
+import { SerializationTools } from './serializationTools';
+import { Observable } from 'babylonjs/Misc/observable';
+/**
+ * Interface used to specify creation options for the gui editor
+ */
+export interface INodeEditorOptions {
+    hostElement?: HTMLElement,
+    customSave?: {label: string, action: (data: string) => Promise<void>};
+    customLoadObservable?: Observable<any>
+}
+
+/**
+ * Class used to create a gui editor
+ */
+export class GuiEditor {
+    private static _CurrentState: GlobalState;
+
+    /**
+     * Show the gui editor
+     * @param options defines the options to use to configure the gui editor
+     */
+    public static Show(options: INodeEditorOptions) {

```

Comment:
Should nt this take at least a scene or engine in parameter ?

Reply:
I don't think so but I found other issues with that line. Like "INodeEditorOptions". Fixed.